### ISSUE TITLE
Static Instance Preload Scene read corruption

### ISSUE THREAD

**Issue by jlturner** [godotengine/godot#81462]
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Godot v4.1.1.stable - macOS 13.5.1 - Vulkan (Compatibility) - AMD Radeon Pro 580X OpenGL Engine - Intel(R) Core(TM) i5-9600K CPU @ 3.70GHz (6 Threads)

### Issue description

https://github.com/godotengine/godot/assets/628713/20ee0edf-0c0a-425d-8ee3-295f0ab8ba91

`static var scene: PackedScene = preload("res://example.tscn")` in a script attached to a `Node` saved as `example.tscn`, when reloaded in Godot, makes the `example.tscn` file unable to be opened in Godot.

It **does** work when running your project however, and if that line is commented out in the script and the project reloaded, then the scene is editable again.

### Steps to reproduce

# Create the following files:

1. `example.gd`:
```
extends Node

static var scene: PackedScene = preload("res://example.tscn")
```

2. `example.tscn`: A scene with a `Node` root with the `example.gd` script attached.

# Reproduce the error
1. Click `Project -> Reload Current Project`
2. Open `example.tscn`
3. Observe error alert `Scene file 'example.tscn' appears to be invalid/corrupt.`
4. Observe console output:
```
scene/resources/resource_format_text.cpp:283 - res://example.tscn:6 - Parse Error: 
  Failed loading resource: res://example.tscn. Make sure resources have been imported by opening the project in the editor at least once.
  scene/resources/packed_scene.cpp:92 - Condition "nc == 0" is true. Returning: nullptr
  editor/editor_data.cpp:880 - Index p_idx = 2 is out of bounds (edited_scene.size() = 2).
  editor/editor_data.cpp:683 - Index p_idx = 2 is out of bounds (edited_scene.size() = 2).
  editor/editor_data.cpp:745 - Index p_idx = 2 is out of bounds (edited_scene.size() = 2).
```

### Minimal reproduction project

[static_scene_bug_repro.zip](https://github.com/godotengine/godot/files/12562791/static_scene_bug_repro.zip)

**Comment by dalexeev** [comment]
This is most likely a duplicate of one of #80877 tracked issues, but maybe there is some unique information. Please check https://github.com/godotengine/godot/issues/74253#issuecomment-1564227775.

**Comment by jlturner** [comment]
Btw for whomever ends up looking into this: 

It seems to occur with any `preload`'ed scene in a `static var` initialization.
`static var test = preload("res://example.tscn")` does it for me

**Comment by MewPurPur** [comment]
Can't reproduce in 4.6

**Comment by dalexeev** [comment]
Indeed, the bug is no longer reproducible. It was likely fixed by #93346 or one of the subsequent PRs.
